internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -897,7 +897,10 @@ func modifyMappedVirtualDisk(
 				return err
 			}
 		}
-		return scsi.UnplugDevice(ctx, mvd.Controller, mvd.Lun)
+		if mvd.UnplugDevice {
```

Comment:
This seems like a breaking change between shim and GCS. Any old shim talking to a new GCS will never have its disks unplugged, because it won't pass this new field.

Reply:
Good point. I can invert it to avoid that since this is a necessary change for this PR. Won't be pretty though :(